subvolumegroup: add support for size and datapool (#14036)
rook/rook: Documentation/CRDs/Shared-Filesystem/ceph-fs-subvolumegroup-crd.md

    distributed: 1            # distributed=<0, 1> (disabled=0)
    # export:                 # export=<0-256> (disabled=-1)
    # random:                 # random=[0.0, 1.0](disabled=0.0)

>   # size of the subvolume group.

Reviewer: "size" is too vague, let's mention it's for a quota
```suggestion
  # Quota size of the subvolume group.
```

  #size: 10G

>   # data pool name for the subvolume group layout.

Reviewer: ```suggestion
  # data pool name for the subvolume group layout instead of the default data pool.
```

* `filesystemName`: The metadata name of the CephFilesystem CR where the subvolume group will be created.

> * `size`: Size of the Ceph Filesystem subvolume group.

Reviewer: ```suggestion
* `size`: Quota size of the Ceph Filesystem subvolume group.
```

> * `dataPoolName`: The data pool name for the Ceph Filesystem subvolume group layout.

Reviewer: ```suggestion
* `dataPoolName`: The data pool name for the Ceph Filesystem subvolume group layout, if the default CephFS pool is not desired. 
```

> * `pinning`: To distribute load across MDS ranks in predictable and stable ways. Reference: <https://docs.ceph.com/en/latest/cephfs/fs-volumes/#pinning-subvolumes-and-subvolume-groups>.

Reviewer: Not related to your changes, but let's use a link instead of raw URL.
```suggestion
* `pinning`: To distribute load across MDS ranks in predictable and stable ways. See the Ceph doc for [Pinning subvolume groups](https://docs.ceph.com/en/latest/cephfs/fs-volumes/#pinning-subvolumes-and-subvolume-groups).
```

>   * `distributed`: Range: <0, 1>, for disabling it set to 0

Reviewer: Did your editor autoformat to only have two leading spaces? IIRC the formatted docs will not have proper indentation if there are not at least three spaces.

Author: yes, fixed it now.

Reviewer: Did you push the fix? It looks the same.

Author: somehow it got reverted but its present in latest push
